src/utp_wrapper.rs

```diff
@@ -47,7 +49,13 @@ impl UtpWrapper {
                                     }
                                 },
                                 Err(TryRecvError::Disconnected) => break 'outer,
-                                Err(TryRecvError::Empty) => break,
+                                Err(TryRecvError::Empty) => {
+                                    if timeout < ::time::SteadyTime::now() {
```

Comment:
Sorry, the link you put seems won't break the outer loop, i.e. stop the thread?
neither from the testing code I see any timeout happens for the outer loop.


The purpose of this PR is to resolve the hanging issue of the test, so the test itself can be included in the CI and to be used to verify the utp related work.
Yes, maybe as you said the change means stop the connection if there is no scheduled msg.
But I think that is design issue to decide whether it is valid or not.
Maybe a keepalive msg is necessary, and I see your work in https://github.com/maidsafe/crust/pull/445 introducing ping-pong msgs.

If you are concerned about the master branch, I will suggest merge this PR into your 445 work, then close this PR.
So PR 445 will contain all the testing code and updates required for the updated rust-utp.